packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -261,26 +264,29 @@ abstract contract CrossDomainMessenger is
 
         if (_isSystemMessageSender()) {
             // Should never happen.
-            require(msg.value == _value, "Mismatched message value.");
+            require(msg.value == _value, "CrossDomainMessenger: Mismatched message value.");
         } else {
-            // TODO(tynes): could require that msg.value == 0 here
-            // to prevent eth from getting stuck
-            require(receivedMessages[versionedHash], "Message cannot be replayed.");
+            require(msg.value == 0, "CrossDomainMessenger: Cannot relay messages with value");
+            require(
+                receivedMessages[versionedHash],
+                "CrossDomainMessenger: Message cannot be replayed."
+            );
         }
 
-        // TODO: Should blocking happen on sending or receiving side?
-        // TODO: Should this just return with an event instead of reverting?
```

Comment:
Either would be secure, but I think reverting on an invalid action is much more standard.